pipeline/sources/webcam_worker.py

```diff
@@ -0,0 +1,225 @@
+
+        jinja2_env = Environment(loader=PackageLoader(package_name='pipeline'), autoescape=select_autoescape())
+        self.index_template = jinja2_env.get_template('webcam_index.html.j2')
+
+    def run(self):
+        # TODO: splitting up in incremental and full download
+        self.download()
+
+        # We always symlink, as we need to publish the latest images
+        self.symlink_with_index()
+
+        # We just clean at full syncs
+        self.clean()
+
+    def download(self):
+        result, _ = self._run_command(
+            command=[
+                'lftp',
+                '-e',
+                f'mirror --newer-than=now-{self.config.keep_days - 1}days -c --parallel={self.config.worker_count} {self.config.remote_dir} {self.config.image_path}; quit;"',
+                '-u',
+                f'{self.config.user},{self.config.password}',
+                self.config.host,
+            ],
+        )
+        match = re.match(r'Total: (\d*) directories?, (\d*) files?, (\d*) symlinks?', result)
+
+        if match is None:
+            self.context.log.error(f'Could not parse lftp output: {result}')
+            return
+
+        directories, files, symlinks = match.groups()
+
+        # TODO: this would be perfect for metrics
+        self.context.log.info(f'Downloaded {files} files, {directories} directories, {symlinks} symlinks')
+
+    def symlink_with_index(self):
+        symlink_items = self.symlink()
+        index_page = self.index_template.render(symlink_items=symlink_items)
+        index_path = Path(self.config.symlink_path, 'index.html')
+        with open(index_path, 'w') as f:
+            f.write(index_page)
+
+    def symlink(self) -> list[SymlinkItem]:
+        image_dir = self.config.image_path
+
+        if not image_dir.exists():
+            return []
+        webcam_base_paths = list(image_dir.iterdir())
+        items: list[SymlinkItem] = []
+        for webcam_base_path in webcam_base_paths:
+            symlink_item = self.symlink_per_webcam_base_path(webcam_base_path)
+
+            if symlink_item is not None:
+                items.append(symlink_item)
+
+        items.sort(key=lambda x: x.name)
+        return items
+
+    def symlink_per_webcam_base_path(self, webcam_base_path: Path) -> SymlinkItem | None:
+        symlink_item = self.get_latest_image_path_per_webcam_base_path(webcam_base_path)
+        if symlink_item is None:
+            return None
+
+        self.config.symlink_path.mkdir(parents=True, exist_ok=True)
+        symlink_path = Path(self.config.symlink_path, symlink_item.filename)
+        symlink_path.unlink(missing_ok=True)
+        os.symlink(symlink_item.path, symlink_path)
+
+        return symlink_item
+
+    def get_latest_image_path_per_webcam_base_path(self, webcam_base_path: Path) -> SymlinkItem | None:
+        # We add 4 hours to make sure that we really get the latest image.
+        check_datetime = datetime.now() + timedelta(hours=4)
```

Comment:
[nitpick]

Currently, this code implicitly assumes that it runs in the same time zone (or at least in one with a similar offset) than the FTP server.

We should either make this assumption more explicit in a comment, or (better) pass `tz` to `datetime.now()` to enforce the `Europe/Berlin` time zone.